core/src/main/java/org/jobrunr/storage/InMemoryStorageProvider.java

```diff
@@ -130,7 +130,7 @@ public int removeTimedOutBackgroundJobServers(Instant heartbeatOlderThan) {
                 .filter(entry -> entry.getValue().getLastHeartbeat().isBefore(heartbeatOlderThan))
                 .map(Map.Entry::getKey)
                 .collect(toList());
-        backgroundJobServers.keySet().removeAll(serversToRemove);
+        serversToRemove.forEach(backgroundJobServers.keySet()::remove);
```

Comment:
I don't think it is on us to fix or workaround this (as the JVM/JDK here is slow and it may be fixed in future releases). 
Also, as the `backgroundJobServers` will always be larger than the `serversToRemove` the issue mentioned by Intellij does not impact us.